x-pack/elastic-agent/docs/elastic-agent-command-line.asciidoc

```diff
@@ -8,40 +10,276 @@
+secure connection.
+
+{global-flags-link}
+
+[discrete]
+=== Example
+
+[source,shell]
+----
+elastic-agent enroll http://localhost:5601 ZnmNIdzVITUJua2QIdU5FTWROVjY6dHY2N1EybWNTMUdPejg5ODbYcVpNUQ==
+----
+
+++++
+<hr>
+++++
+
+[discrete]
+[[elastic-agent-help-command]]
+== elastic-agent help
+
+Show help for a specific command.
+
+[discrete]
+=== Synopsis
+
+[source,shell]
+----
+elastic-agent help <command> [--help] [global-flags]
+----
+
+[discrete]
+=== Options
+
+`command`::
+The name of the command.
+
+`--help`::
+Show help for the `help` command.
+
+{global-flags-link}
+
+[discrete]
+=== Example
+
+[source,shell]
+----
+elastic-agent help enroll
+----
+
+++++
+<hr>
+++++
+
+[discrete]
+[[elastic-agent-inspect-command]]
+== elastic-agent inspect
+
+Show the current {agent} configuration.
+
+If no parameters are specified, shows the full {agent} configuration.
+
+[discrete]
+=== Synopsis
+
+[source,shell]
+----
+elastic-agent inspect [--help] [global-flags]
+elastic-agent inspect output [--output <string>] [--program <string>]
+                             [--help] [global-flags]
+----
+
+
+//REVIEWERS: The language used to describe this option is a little confusing. 
+//Beats users are likely to think of "output" as the output section for sending
+//to Elasticsearch or Logstash. Any suggestions for making this more clear?
+
+[discrete]
+=== Options
+
+`output`:: Display the configuration generated for the output. This command
+accepts additional flags:
 +
-If not specified, {agent} uses current working directory.
+--
+`--output <string>`::
+The name of the output to introspect.
+
+`--program <string>`::
+The type of program to introspect. For example, `filebeat`. This option must be
+combined with `--output`. 
+--
+
+`--help`::
+Show help for the `inspect` command.
+
+{global-flags-link}
+
+[discrete]
+=== Examples
+
+[source,shell]
+----
+elastic-agent inspect
+elastic-agent inspect output --output default
+elastic-agent inspect output --output default --program filebeat
+----
+
+++++
+<hr>
+++++
+
+[discrete]
+[[elastic-agent-run-command]]
+== elastic-agent run
+
+Start the `elastic-agent` process.
+
+[discrete]
+=== Synopsis
+
+[source,shell]
+----
+elastic-agent run [global-flags]
+----
+
+[discrete]
+[[elastic-agent-global-flags]]
+=== Global flags
+
+These flags are valid whenever you run `elastic-agent` on the command line.
+
+`-c <string>`::
+The configuration file to use. If not specified, {agent} uses
+`{path.home}/elastic-agent.yml`.
+
+`-d <string>`::
+Enable the specified debug selectors.
+
+//REVIEWERS: If we document the -d flag, we really should document selectors.
+//Which selectors are available with agent?
 
-`--path.data`::
-The data directory used by {agent} to store downloaded artifacts. Also stores
-logs for any {beats} started and managed by {agent}.
+`--e`::
+Log to stderr and disable syslog/file output.
+
+`--environment <environmentVar>`::
```

Comment:
@ph Could you comment on this? I would have to dig into the code.

Reply:
I'll add a to-do to follow up...not sure PH will have time to respond before I merge this.